helmctlr/helm.go

```diff
@@ -118,6 +118,16 @@ func (c Controller) installOrUpdate(r *unstructured.Unstructured) error {
 	if err != nil {
 		return err
 	}
+
+	// If chart entry is present in CR, then download the chart from repo to local dir
+	// on any error, fallback to local chart already present on the disk
+	chartRef := getChartRef(r)
+	if chartRef != "" {
+		if chartDir, chartErr := getRemoteChart(chartRef); chartErr == nil {
+			c.ChartDir = chartDir
```

Comment:
I'm not sure I agree with this decision. If a remote chart is specified it is clear the user wants that chart to be deployed, not the local one. I don't think we should default to the local chart here. 

Either way, no matter which approach we take there should be logs to call out it failed.